File: python/balance_parens.py
import re
def balance_parens(str):
    strL = list(str)
    stack = []
    # print(strL)

    for i in range(len(str)):
        if str[i] == "(":
            stack.append(i)
        if str[i] == ")":
            if len(stack) == 0:
                strL[i] = ""
            else:
                stack.pop()
    for j in stack:
        strL[j] = ""
    result= ''.join(strL)
    return result
    

File: js/balanceParens.js
balanceParens = (str) =>
{
    let splitted = str.split("")
    let stack = []

    for (let i = 0; i < str.length; i++) {
        if (splitted[i] === "("){
            stack.push(i)
        } else if ( splitted[i] === ")"){
            if (stack.length == 0){
                splitted[i] = ""
            } else{
                stack.pop()
            }
        }
    }
    for (j of stack) {
        splitted[j] = ""
    }
    // console.log(splitted)
    return splitted.join("")
}

module.exports = { balanceParens }